### ISSUE TITLE
[Feature] Allow Nicknaming Remote Device

### ISSUE THREAD

**Issue by Lindsor** [ProxymanApp/Proxyman#466]
### Description

Currently Proxyman allows you to proxy a remote device (ie. Android) through Proxyman's proxy and inspect those calls.

The remote device displays as its local IP address (ie. 10.1.2.3)
It would be nice to be able to give a nickname to the connected device, something like:
`Android Pixel (10.1.2.3)`

The only issue I can see with this is if the local IP's change then the mapping breaks.
However, in my opinion, people using a feature like this would be Tech Savvy enough to understand this limitation, also why I added the `(<IP_ADDRESS>)` after the nickname.

![image](https://user-images.githubusercontent.com/3959304/78158095-d2030980-740e-11ea-8dc3-bd5cae627668.png)

**Comment by NghiaTranUIT** [comment]
Hi @Lindsor, it's actually a reasonable idea since I sometimes struggle to guess which the device is when I have two connected devices. 

I would introduce a Set Nick Name in the right-context menu, which allows us to set a name (with a simple caching mechanism too). The format might be "<NICKNAME> (<IP_ADDRESS>)" like your suggestion 👍 

I would support it in the next release 🌮 Thank you for this convenience idea 🐝

**Comment by yauheniprakapenka** [comment]
i also want this feature too! thanks

**Comment by NghiaTranUIT** [comment]
Good new 🌮 @yauheniprakapenka @Lindsor 
BETA: https://proxyman.s3.us-east-2.amazonaws.com/beta/Proxyman_1.20.0_Nicknam_Remote_Device.dmg

- You can set a nickname directly on the cell (Same behavior with Rename Folder)
- Simple Caching by the device's IP. So if you change the network, the name could be changed.

<img width="402" alt="Screen Shot 2020-04-05 at 12 36 33" src="https://user-images.githubusercontent.com/5878421/78468617-dba8ac00-7743-11ea-90c5-e254fd49e416.png">
<img width="542" alt="Screen Shot 2020-04-05 at 12 38 30" src="https://user-images.githubusercontent.com/5878421/78468625-e2cfba00-7743-11ea-8bda-c91ef6db594a.png">
<img width="290" alt="Screen Shot 2020-04-05 at 12 36 38" src="https://user-images.githubusercontent.com/5878421/78468628-e5caaa80-7743-11ea-9f31-f6fcc3494a61.png">

**Comment by Lindsor** [comment]
Amazing, very quick turn around, you rock

**Comment by crankygeek** [comment]
I love this soooo much. I've been meaning to add a request for something like this. I typically have 5+ devices proxying and Remote Devices has mostly been unusable for me.

Thanks @Lindsor and @NghiaTranUIT!!

**Comment by NghiaTranUIT** [comment]
Just friendly reminder that this feature is rolled out at the latest build 1.21.0 🙌 

Thank you all 🙇

**Comment by crankygeek** [comment]
Yup, and it works really well. Now I just need to get my enhancement requests in. ;)

**Comment by chornerman** [comment]
Hi, @NghiaTranUIT 👋🏼  Is it possible to see a list of all of the nicknames I've set.

**Comment by crankygeek** [comment]
I don't think the list is exposed in the UI, but the info is stored in ~/Library/Preferences/com.proxyman.NSProxy.plist under `RemoteDeviceNickName`.

**Comment by NghiaTranUIT** [comment]
Thanks @crankygeek. The list name is in this file @chornerman 👍 

Please take note that Proxyman doesn't know which device model is connecting to Proxyman. It just a map of a Device's IP and the Nickname.

**Comment by chornerman** [comment]
Thanks, @crankygeek @NghiaTranUIT 🙇🏼‍♂️
